### PR TITLE
Bug/42 search ampersands apostrophes

### PR DESCRIPTION
**Pull request by rossettistone**
Includes bump to `acorn` lib, which is unrelated but (as a dependency of `webpack`) was tested in the course of preparing this PR.

Tested with a few different search terms. One that might be helpful for review: "O'Reilly & Co."

### CODE REVIEW

File: src/search/index.js [hunk 1]
```diff
@@ -184,11 +184,11 @@ export default class CoProcureSearch extends HTMLElement {
     if(this.query || this.prepop) {
       if(this.query && this.query.indexOf('"')<0) {
         url += `&q=(and `;
-        decodeURIComponent(this.query).split(' ').forEach( (term) => {
-          url += ` '${term}'`;
+        this.query.split(' ').forEach( (term) => {
```
Comment (achen530) [inline src/search/index.js:187]: So just checking, this logic here was to do exact phrase searches when there were quotes on a term, and that still works?

File: src/search/index.js [hunk 2]
```diff
@@ -423,6 +423,15 @@ export default class CoProcureSearch extends HTMLElement {
     document.head.appendChild(script2);
   }
 
+  prepareSearchQueryTerm(term) {
+    // AWS CloudSearch chokes on single quotes in queries unless they're escaped.
+    // Escaping backslashes must be added *after* URI encoding, otherwise they'll be cleaned out.
+    let preparedTerm = encodeURIComponent(term)
+    preparedTerm = preparedTerm.replace(/'/g, "%5C'");
```
Comment (achen530) [inline src/search/index.js:430]: Does encodeURIComponent take care of the backslashes?

**Review by achen530 — APPROVED** [review]

File: src/search/index.js (hunk 1 above)
Comment (rossettistone) [reply src/search/index.js:187]: Excellent question. Using the string "breath testing," I observed that the string (with quotes) gets the same number of results locally and on prod *and* that there are somewhat more results if searching without quotes around the string.

TL;DR: Yep, I think it's still working properly.

File: src/search/index.js (hunk 2 above)
Comment (rossettistone) [reply src/search/index.js:430]: `encodeURIComponent` will convert existing backslashes to %5C, but in this case I'm trying to add *new* backslashes in front of all apostrophes. Another approach would be to do this above the `encodeURIComponent` call, but I think I was having trouble getting the backslashes into the URLs (since they're also escape characters), so I ultimately found it simpler to just prepend the already-converted `%5C`s instead.

Does that seem alright?
